internal/config/config.go

```diff
@@ -0,0 +1,252 @@
+package config
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"sync"
+
+	"github.com/google/uuid"
+	"github.com/lestrrat-go/jwx/jwt"
+)
+
+// ErrConfigFileMissing is thrown when the config.json file is missing.
+var ErrConfigFileMissing = errors.New("config.json file is missing")
+
+// ErrNoAuthenticatedTenants is thrown when the config file has no authenticated tenants.
+var ErrNoAuthenticatedTenants = errors.New("Not logged in. Try `auth0 login`.")
+
+// Config holds cli configuration settings.
+type Config struct {
```

Comment:
Should we expose an interface with `Config`'s methods, and then use the interface in the `cli` [struct](https://github.com/auth0/auth0-cli/blob/main/internal/cli/cli.go#L71), so that the config can be mocked? It would be useful for testing functions that use config methods like `GetTenant()`, such as this one: https://github.com/auth0/auth0-cli/blob/main/internal/cli/apps.go#L882

Reply:
IMO it's unnecessary as right now we can simply inject an in-memory Config struct with any values we might need for testing.

So the setup to test for example the `appPickerOptions` func would be similar to:

https://github.com/auth0/auth0-cli/blob/de878571100332e13bab5abe65af973611a34ca3/internal/config/config_test.go#L206-L234

Reply:
That involves creating an actual temp config file. But it certainly does the trick.